### ISSUE TITLE
IPython REPL doesn't render exception notes

### ISSUE THREAD

**Issue by wimglenn** [ipython/ipython#13849]
Adding additional context to exceptions is possible [in Python 3.11+](https://docs.python.org/3/library/exceptions.html#BaseException.add_note), described in [_PEP 678 – Enriching Exceptions with Notes_](https://peps.python.org/pep-0678/).

In vanilla REPL:

```
>>> try:
...     1/0
... except Exception as e:
...     e.add_note("uh-oh")
...     raise
... 
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
ZeroDivisionError: division by zero
uh-oh
```

They don't show in the IPython REPL:

```
In [1]: try:
   ...:     1/0
   ...: except Exception as e:
   ...:     e.add_note("uh-oh")
   ...:     raise e
   ...: 
---------------------------------------------------------------------------
ZeroDivisionError                         Traceback (most recent call last)
Cell In[1], line 5
      3 except Exception as e:
      4     e.add_note("uh-oh")
----> 5     raise e

Cell In[1], line 2
      1 try:
----> 2     1/0
      3 except Exception as e:
      4     e.add_note("uh-oh")

ZeroDivisionError: division by zero

In [2]: import sys

In [3]: sys.last_value.__notes__
Out[3]: ['uh-oh']

```

**Comment by ivirshup** [comment]
I additionally don't see notes in the notebook or lab interfaces.

**Comment by ivirshup** [comment]
When this is fixed in IPython, it would be useful if it worked on python versions < 3.11. This is currently achievable in the vanilla repl by importing [`exceptiongroup`](https://github.com/agronholm/exceptiongroup), which patches the default traceback, and adding a `__notes__` attribute to Exceptions.

**Comment by ivirshup** [comment]
I believe this was fixed by:

* https://github.com/ipython/ipython/pull/14039

**Comment by flying-sheep** [comment]
That means it’s fixed in [8.14.0](https://github.com/ipython/ipython/releases/tag/8.14.0)

**Comment by Zac-HD** [comment]
@Carreau I believe this has been fixed 🙂
